src/backend/src/services/teams.services.ts

```diff
@@ -152,8 +159,20 @@ export default class TeamsService {
 
     if (team.dateArchived) throw new HttpException(400, 'Cannot edit the head of an archived team');
 
-    if (newHead && team.members.map((user) => user.userId).includes(newHead?.userId))
-      throw new HttpException(400, 'Error: Team head cannot be a member');
+    // if the new head is a current member on the team, this will remove them as a member
+    if (newHead && team.members.map((user) => user.userId).includes(newHead?.userId)) {
+      const newTeamArr: number[] = team.members
+        .filter((member) => member.userId !== newHead.userId)
+        .map((member) => member.userId);
+      this.setTeamMembers(submitter, teamId, newTeamArr);
+    }
+
+    // if the new head is a current leader on the team, they will be removed as a leader
+    // this will work in theory, but there is a block on the front-end that prevents the user from adding a current lead as the head of a team (they won't show up in the new head dropdown)
+    if (team.leads.map((lead) => lead.userId).includes(userId)) {
```

Comment:
should this be .includes(newHead?.userId)

Reply:
and also do you need newHead &&?

Reply:
> should this be .includes(newHead?.userId)

If I'm not wrong, I think userId is the parameter that's passed into the function which represents the userId of the new Head, so saying newHead?.userId might be redundant? But if that's incorrect I'll definitely change that. 


> and also do you need newHead &&?

yeah I think I do, I just forgot that. I'll go ahead and fix that now.

Reply:
As far as using newHead?, you use it on line 163, but don't use it on line 172, and I think they should be the same? If it isn't necessary, can it be removed from line 163?

Reply:
Oh yeah good point, I'll remove it now.